NewsDistribution/NewsClient.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net.Sockets;
+
+namespace NewsDistribution;
+
+public class NewsClient
+{
+    private Socket? _socket;
+    private NetworkStream? _stream;
+    private readonly byte[] _buffer = new byte[2048];
+    Thread? _receiveThread;
+
+    public delegate void NewsReceived(string title, string description, string content);
+    public event NewsReceived? OnNewsReceived;
+
+    public bool Connect(string name, string address, ushort port)
+    {
+        if (name == string.Empty || name.Length > 256)
+            throw new ArgumentException("Name is either empty or too long (>256).", nameof(name));
+
+        if (_socket != null)
+            return false;
+
+        try
+        {
+            _socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            _socket.Connect(address, port);
+        }
+        catch (SocketException)
+        {
+            _socket?.Dispose();
+            _socket = null;
+
+            return false;
+        }
+
+        _stream = new(_socket);
+
+        SendAuthorizationPacket(name);
+        bool authorized = ReceiveAuthorizationPacket();
+
+        if (authorized)
+        {
+            _receiveThread = new Thread(ReceiveThreadProc);
+            _receiveThread.Start();
+        }
+        else
+        {
+            _stream.Close();
+            _socket.Close();
+            _stream = null;
+            _socket = null;
+        }
+
+        return authorized;
+    }
+
+    public void Disconnect(bool sendPacket = true)
+    {
+        if (_socket == null)
+            return;
+
+        if (sendPacket)
+        {
+            _buffer[0] = (byte)PacketType.Unsubscribe;
+            _socket.Send(_buffer, 1, SocketFlags.None);
+        }
+
+        _socket.Disconnect(false);
+        _stream!.Close();
+        _socket = null;
+    }
+
+    private void ReceiveThreadProc()
+    {
+        if (_socket == null)
+            return;
+
+        while (true)
+        {
+            _socket.Receive(_buffer, 1, SocketFlags.None);
+
+            switch ((PacketType)_buffer[0])
```

Comment:
An exception is required for the default case.
For example:
```
default:
    throw new ArgumentOutOfRangeException();
```

Reply:
It would be exploitable by sending specially crafted packets with invalid packet types.